src/components/Toast/ToastContext.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { createContext, PropsWithChildren, ReactNode, useContext, useMemo, useState } from "react";
+import { Toast, ToastTypes } from "./Toast";
+
+export interface ToastNoticeProps {
+  type: ToastTypes;
+  message: ReactNode;
+}
+
+export type ToastContextProps = {
+  notice: ToastNoticeProps | undefined;
+  setNotice: React.Dispatch<React.SetStateAction<ToastNoticeProps | undefined>>;
+};
+
+export const ToastContext = createContext<ToastContextProps>({
+  setNotice: () => {
+    throw new Error("Missing ToastProvider");
+  },
+  notice: { type: "error", message: "" },
+});
+
+export function ToastProvider(props: PropsWithChildren<{}>) {
+  const [notice, setNotice] = useState<ToastNoticeProps>();
+  const contextValue = useMemo(() => ({ setNotice, notice }), [setNotice, notice]);
+  return (
+    <ToastContext.Provider value={contextValue}>
+      {props.children}
+      <Toast />
```

Comment:
**request:** Can we remove `<Toast />` here? The application should decided where the `Toast` component will be placed. Beam only needs to provide the ToastContext wrapper around the application.

Reply:
When doing this, you'll need to update the `useToast` hook's `TestToastComponent` to be something like:
```
function TestToastComponent() {
  return (<>
   // Include the Toast component here.
    <Toast />
    <Button onClick={...} />
  </>);
}